horace_core/sqw/coord_transform/@ortho_proj/private/build_from_old_data_struct_.m

```diff
@@ -16,22 +9,35 @@
 if isfield(data_struct,'ulabel')
     data_struct.label = data_struct.ulabel;
 end
+if isfield(data_struct,'uoffset')
+    data_struct.offset = data_struct.uoffset;    
+end
+if isfield(data_struct,'u_to_rlu')
+    use_u_to_rlu_transitional = true;
+else
+    use_u_to_rlu_transitional = false;    
+end
 
-proj=ortho_proj('alatt',alatt,'angdeg',angdeg,'label',data_struct.label);
-%
 
-proj = proj.set_from_data_mat(data_struct.u_to_rlu(1:3,1:3),data_struct.ulen(1:3));
-if isfield(data_struct,'offset')
-    proj.offset = data_struct.offset;
-elseif isfield(data_struct,'uoffset')
-    proj.offset = data_struct.uoffset;
+flds = fieldnames(data_struct);
+avail_fields = proj.saveableFields();
+valid = ismember(flds,avail_fields);
```

Comment:
Should probably warn if any aren't valid

Reply:
reduces to `from_bare_struct` method